### ISSUE TITLE
Make the default image used by capz based on ubuntu 20.04 

### ISSUE THREAD

**Issue by nader-ziada** [kubernetes-sigs/cluster-api-provider-azure#1447]
/kind feature

**Describe the solution you'd like**
Currently the default image used by capz when no user image is supplied is based on `ubuntu 18.04` which is getting old and doesn't have all the patches back-ported to it. This default image is also used by e2e tests.
Should publish and use `20.04` images

The images published as of the time this issue is created are the following:
```
% az vm image list --publisher cncf-upstream --offer capi --all -o table --query "[?version=='2021.05.17']"
Publisher      Offer         Sku                           Version     Urn
-------------  ------------  ----------------------------  ----------  ------------------------------------------------------------------
cncf-upstream  capi          k8s-1dot18dot19-ubuntu-1804   2021.05.17  cncf-upstream:capi:k8s-1dot18dot19-ubuntu-1804:2021.05.17
cncf-upstream  capi          k8s-1dot19dot11-ubuntu-1804   2021.05.17  cncf-upstream:capi:k8s-1dot19dot11-ubuntu-1804:2021.05.17
cncf-upstream  capi          k8s-1dot20dot7-ubuntu-1804    2021.05.17  cncf-upstream:capi:k8s-1dot20dot7-ubuntu-1804:2021.05.17
cncf-upstream  capi          k8s-1dot21dot1-ubuntu-1804    2021.05.17  cncf-upstream:capi:k8s-1dot21dot1-ubuntu-1804:2021.05.17
cncf-upstream  capi-windows  k8s-1dot18dot19-windows-2019  2021.05.17  cncf-upstream:capi-windows:k8s-1dot18dot19-windows-2019:2021.05.17
cncf-upstream  capi-windows  k8s-1dot19dot11-windows-2019  2021.05.17  cncf-upstream:capi-windows:k8s-1dot19dot11-windows-2019:2021.05.17
cncf-upstream  capi-windows  k8s-1dot20dot7-windows-2019   2021.05.17  cncf-upstream:capi-windows:k8s-1dot20dot7-windows-2019:2021.05.17
cncf-upstream  capi-windows  k8s-1dot21dot1-windows-2019   2021.05.17  cncf-upstream:capi-windows:k8s-1dot21dot1-windows-2019:2021.05.17
```

**Anything else you would like to add:**
- images build by [image builder](https://github.com/kubernetes-sigs/image-builder/tree/master/images/capi)

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by mboersma** [comment]
> Should publish and use 20.04 images

👍🏻 I think so, 2018 was like a thousand years ago. We could have a few releases where we publish both 18.04 and 20.04, then maybe drop 18.04 to simplify things?

**Comment by nader-ziada** [comment]
> > Should publish and use 20.04 images
> 
> 👍🏻 I think so, 2018 was like a thousand years ago. We could have a few releases where we publish both 18.04 and 20.04, then maybe drop 18.04 to simplify things?

+1 to publishing both then eventually switching

**Comment by mboersma** [comment]
We published 20.04 reference images for the recent Kubernetes patches (1.18.20, 1.19.12, 1.20.8, and 1.21.2), and we will publish both 20.04 and 18.04 images each month. (Maybe until April 2022, when 18.04 goes out of support.)

But how should we handle the default choice for Ubuntu in CAPZ? I can think of 3 options:

1. Include specific "if >= patch version" tests in `GetDefaultUbuntuImage` that know what we've published as reference images
2. Always try 20.04 first and somehow handle the error and fall back to 18.04 if it wasn't found
3. Retroactively publish 20.04 reference images for those versions that don't have them, so the code can just default to 20.04

I don't love any of these, frankly, but would probably go with 1). But I'd love to hear fresh ideas or any feedback anyone has.
